### ISSUE TITLE
Who keeps track of IPosition? 

### ISSUE THREAD

**Issue by JeroenBos** [JeroenBos/ASDE#13]
Suppose we have a box containing some text, in the middle of which resides a box composition containing a single box. 
![untitled](https://cloud.githubusercontent.com/assets/15310631/16472898/af9e5226-3e68-11e6-856b-017932b3617f.png)

The position of the letters are obviously linear positions, and hence that of the box composition must also be a linear position. But the box composition itself doesn't know that (well, any box composition always has a linear position as position, but *which* is unknown to it). The box in the box composition has a matrix position (0, 0). Again, that box doesn't know that. It could have been a box in some other composition or at another spot in a matrix. 

Basically, what I'm trying to get at is that a position of an element depends on where it is used, i.e. is an extrinsic property. It therefore doesn't make sense for that element to determine its own position. 

The Model
The box is a AtomicNameInstance with position (0, 0). The box composition is a CompositeNameInstance with the above as single element and position Linear(2). But it doesn't know that yet until it is placed there. We could give it position 'Root', but that would then at some time be replaced by Linear(2). 

What is a correct design here? Immutability of the name instances should also still be true...

**Comment by JeroenBos** [comment]
Current implementation has mutable name instances, *but* during name inference they are not allowed to mutate

**Comment by JeroenBos** [comment]
So box elements don't determine their own positions, the owning box regulates that.
Similarly, a box composition determines the positions of its boxes.

**Comment by JeroenBos** [comment]
Does a box model have a position?

Well, if you think about a box as part of a box composition, then yes, a box has a position. The question is whether it should be stored in a parallel collection in the box composition, or on the boxes themselves. 

I think it's simpler and more versatile to just store it on the boxes themselves.

**Comment by JeroenBos** [comment]
For the root box it would be preferable to store the positions on the boxes themselves, such that it can have the root position without having a box composition parent.
